### PR TITLE
Chainsaw & Chainsword update and Add chainsaw to req

### PR DESCRIPTION
**Pull request by AusarAtkin**
## About The Pull Request

Hopefully update and make the chainsaw and chainsword better, now it has a fuel system, attack loop for attacking object like doors/xeno structure, BUT not turf like resin walls (yet at least) and be able to be bought in req for 500 points. 

https://github.com/user-attachments/assets/60dbf949-0d9d-4291-b24e-96f69808ec25

The numbers are still up to change, and up to ridgyax and the maints, first attempt at adding a weapon, prepare your eyes maints.

Also made some change to dripping component so it can use something other than neon blue tracker drip.

## Why It's Good For The Game

More content, and be a gap for melee user, where they don't care about the stun that rocket sledge has and only want the damage part.
## Changelog
:cl:
add: added chainsaw to req for 500 points
balance: chainsaw now use fuel and do more damage while having a rev up delay.
/:cl:

### CODE REVIEW

**Comment by Runian** [comment]
Rocket Sledge is 600 req points, but does 150 damage with a stun.
Chainsaw is 500 req points, but does 125 damage.
Vali Claymore is free, but does 90 damage and has a slower attack speed (24 compared to 20).

I don't feel like it fills the damage-only niche when arguably it is worst than its better alternative for 100 more points. Maybe make it cheaper and raise up the damage a bit more?

**Comment by AusarAtkin** [comment]
> Rocket Sledge is 600 req points, but does 150 damage with a stun.
> Chainsaw is 500 req points, but does 125 damage.
> Vali Claymore is free, but does 90 damage and has a slower attack speed (24 compared to 20).
> 
> I don't feel like it fills the damage-only niche when it is arguably worse than its alternatives. Maybe make it cheaper and raise up the damage a bit more?

Since this is an attempt at bounty, its up to ridgyax, but probably gonna change up the numbers when maint deems the code is good enough to merge.

**Comment by zzurge** [comment]
The emperor protects

**Review by Kuro020 — CHANGES_REQUESTED** [review]
I dont see why exactly marines need another melee weapon? Vali covers the free weapon slot, sledge is a upgrade. So what is the purpose of this one?

**Comment by AusarAtkin** [comment]
> I dont see why exactly marines need another melee weapon? Vali covers the free weapon slot, sledge is a upgrade. So what is the purpose of this one?

It has the damage part of sledge (that actually fully get affected by melee skill so synth and researcher dont benefit from it more than other weapon). But with no cancer part that is the hard stun.

So we gave it 100 less cost. It has loud ass wielding channel to turn it on (and get the damage bonus) and i add a bloodtrail effect as a funny cosmetic bonus (because cool). 

And as ridgyax said, chainsaw cool, he even wanted to give this a chance to delimb fellow humans. And we are willing to change the numbers or some features since essentially ridgyax just want to use chainsaw.

**Review by Kuro020 — APPROVED** [review]

File: code/game/objects/items/weapons/twohanded.dm [hunk 1]
```diff
@@ -608,3 +609,211 @@
...
+		update_item_state()
+	else
+		force = initial(force)
+		hitsound = initial(hitsound)
+		if(reagents.get_reagent_amount(/datum/reagent/fuel) < fuel_used)
+			to_chat(user, span_warning("\The [src] doesn't have enough fuel!"))
+			return
+		to_chat(user, span_warning("\The [src]'s motor died down!"))
+		update_icon()
+		update_item_state()
```
Comment (TiviPlus) [inline code/game/objects/items/weapons/twohanded.dm:686]: early returns, use balloons pls

File: code/game/objects/items/weapons/twohanded.dm [hunk 2]
```diff
@@ -608,3 +609,211 @@
...
+	AddElement(/datum/element/strappable) //Chainsaw can be strapped on hand
+
+///handle icon change
+/obj/item/weapon/twohanded/chainsaw/update_icon_state()
+	. = ..()
+	if(active) //weapon is active
+		icon_state = icon_state_on
+		return
+	else
+		icon_state = initial(icon_state)
```
Comment (TiviPlus) [inline code/game/objects/items/weapons/twohanded.dm:655]:
```suggestion
	icon_state = initial(icon_state)
```

File: code/game/objects/items/weapons/twohanded.dm [hunk 3]
```diff
@@ -608,3 +609,211 @@
...
+/obj/item/weapon/twohanded/chainsaw/update_item_state(mob/user)
+	. = ..()
+	if(active) //weapon is active
+		worn_icon_state = worn_icon_state_on
+		return
+	if(CHECK_BITFIELD(item_flags, WIELDED)) // weapon is wielded but off
+		worn_icon_state = worn_icon_state_w
+		return
+	else
+		worn_icon_state = initial(worn_icon_state)
```
Comment (TiviPlus) [inline code/game/objects/items/weapons/twohanded.dm:667]:
```suggestion
	worn_icon_state = initial(worn_icon_state)
```

File: code/game/objects/items/weapons/twohanded.dm [hunk 4]
```diff
@@ -608,3 +609,211 @@
...
+	reagents.remove_reagent(/datum/reagent/fuel, fuel_used) //remove fuel by fuel_used amount
+	user.changeNext_move(attack_speed) //this is here because attacking object for some reason dont respect weapon attack speed
+	if(reagents.get_reagent_amount(/datum/reagent/fuel) < fuel_used && active) //turn off the chainsaw after one last attack when fuel ran out
+		playsound(loc, 'sound/items/weldingtool_off.ogg', 50)
+		to_chat(user, span_warning("\The [src] shuts off, using last bits of fuel!"))
+		active = FALSE
+		toggle_motor(user)
+		return
+	if(prob(1)) // small chance for an easter egg of simpson chainsaw noises
+		playsound(loc, 'sound/weapons/chainsaw_simpson.ogg', 60)
```
Comment (TiviPlus) [inline code/game/objects/items/weapons/twohanded.dm:701]:
```suggestion
	if(prob(0.01)) // small chance for an easter egg of simpson chainsaw noises
		playsound(loc, 'sound/weapons/chainsaw_simpson.ogg', 60)
```
you underestimate how common 1% is .1 might be ok too

File: code/game/objects/items/weapons/twohanded.dm [hunk 5]
```diff
@@ -608,3 +609,211 @@
...
+			return ..()
+
+		var/fuel_transfer_amount = min(saw.reagents.total_volume, (max_fuel - reagents.get_reagent_amount(/datum/reagent/fuel)))
+		saw.reagents.remove_reagent(/datum/reagent/fuel, fuel_transfer_amount)
+		reagents.add_reagent(/datum/reagent/fuel, fuel_transfer_amount)
+		playsound(loc, 'sound/effects/refill.ogg', 25, 1, 3)
+		to_chat(user, span_notice("You refill [src] with fuel."))
+		update_icon()
+
+	return ..()
```
Comment (TiviPlus) [inline code/game/objects/items/weapons/twohanded.dm:756]: early returns, balloons

File: code/game/objects/items/weapons/twohanded.dm [hunk 6]
```diff
@@ -608,3 +609,211 @@
...
+
+///extra effect when attacking mob, such as xeno and human
+/obj/item/weapon/twohanded/chainsaw/attack(mob/living/carbon/M as mob, mob/living/carbon/user as mob)
+	rip_apart(user)
+	if(!active) //no extra effect when weapon is off
```
Comment (TiviPlus) [inline code/game/objects/items/weapons/twohanded.dm:761]: as mob? 
also you're leaving a lot of unnecessary comments, a person can read that active doesnt do anything and etc

File: code/game/objects/items/weapons/twohanded.dm [hunk 7]
```diff
@@ -608,3 +609,211 @@
...
+		return ..()
+
+	if(isxeno(M))
+		M.AddComponent(/datum/component/dripping, DRIP_ON_WALK, 6, 6, /obj/effect/decal/cleanable/blood/xeno) //leave pool of blood
+		return ..()
+
+	if(ishuman(M))
+		M.AddComponent(/datum/component/dripping, DRIP_ON_WALK, 6, 6, /obj/effect/decal/cleanable/blood) //leave pool of blood
```
Comment (TiviPlus) [inline code/game/objects/items/weapons/twohanded.dm:769]: does this respect synth white blood

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/game/objects/items/weapons/twohanded.dm (hunk 6 above)
Comment (AusarAtkin) [reply code/game/objects/items/weapons/twohanded.dm:761]: To be completely honest, i took that from rocket sledge, and now i think about it it's kinda pointless, ill remove it.
Tbh i started to add comments to help my brain in testing things, ill clean it up on the irrelevant stuffs.

File: code/game/objects/items/weapons/twohanded.dm (hunk 4 above)
Comment (AusarAtkin) [reply code/game/objects/items/weapons/twohanded.dm:701]: Oky

File: code/game/objects/items/weapons/twohanded.dm (hunk 3 above)
Comment (AusarAtkin) [reply code/game/objects/items/weapons/twohanded.dm:667]: huh, why i use else there, ok

File: code/game/objects/items/weapons/twohanded.dm (hunk 2 above)
Comment (AusarAtkin) [reply code/game/objects/items/weapons/twohanded.dm:655]: ok

File: code/game/objects/items/weapons/twohanded.dm (hunk 1 above)
Comment (AusarAtkin) [reply code/game/objects/items/weapons/twohanded.dm:686]: I swear something went wrong here when i tried using early returns then i went back to this, ill tinker with it again for a bit. (Pain of 1 month gap between coding and review)

File: code/game/objects/items/weapons/twohanded.dm (hunk 7 above)
Comment (AusarAtkin) [reply code/game/objects/items/weapons/twohanded.dm:769]: Oh yea i forgot about the tincans gang

**Pull request merged**